src/hooks/useANJ.js

```diff
@@ -21,81 +21,102 @@ import { useConnectedAccount } from '../providers/Web3'
 export function useANJBalances() {
   const { balances, movements } = useDashboardState()
 
+  const {
+    walletBalance,
+    inactiveBalance,
+    activeBalance,
+    lockedBalance,
+    deactivationBalance,
+  } = balances || {}
+
   const convertedMovements = useConvertedMovements(movements)
 
-  const walletBalance = useBalanceWithMovements(
-    balances.walletBalance,
+  const convertedWalletBalance = useBalanceWithMovements(
+    walletBalance,
     convertedMovements,
     anjBalanceTypes.Wallet
   )
 
-  const inactiveBalance = useBalanceWithMovements(
-    balances.inactiveBalance,
+  const convertedInactiveBalance = useBalanceWithMovements(
+    inactiveBalance,
     convertedMovements,
     anjBalanceTypes.Inactive
   )
 
-  const activeBalance = useBalanceWithMovements(
-    balances.activeBalance,
+  const convertedActiveBalance = useBalanceWithMovements(
+    activeBalance,
     convertedMovements,
     anjBalanceTypes.Active
   )
 
-  const lockedBalance = useMemo(() => {
-    return { amount: balances.lockedBalance }
-  }, [balances.lockedBalance])
+  const convertedLockedBalance = useMemo(() => {
+    return { amount: lockedBalance }
+  }, [lockedBalance])
 
-  const deactivationBalance = useMemo(() => {
-    return { amount: balances.deactivationBalance }
-  }, [balances.deactivationBalance])
+  const convertedDeactivationBalance = useMemo(() => {
+    return { amount: deactivationBalance }
+  }, [deactivationBalance])
 
   // Since we pass the whole object through props to components, we should memoize it
-  return useMemo(
-    () => ({
-      walletBalance,
-      inactiveBalance,
-      activeBalance,
-      lockedBalance,
-      deactivationBalance,
-    }),
-    [
-      activeBalance,
-      deactivationBalance,
-      inactiveBalance,
-      lockedBalance,
-      walletBalance,
-    ]
-  )
+  return useMemo(() => {
+    if (!balances) {
+      return null
+    }
+
+    return {
+      walletBalance: convertedWalletBalance,
+      inactiveBalance: convertedInactiveBalance,
+      activeBalance: convertedActiveBalance,
+      lockedBalance: convertedLockedBalance,
+      deactivationBalance: convertedDeactivationBalance,
+    }
+  }, [
+    balances,
+    convertedActiveBalance,
+    convertedDeactivationBalance,
+    convertedInactiveBalance,
+    convertedLockedBalance,
+    convertedWalletBalance,
+  ])
 }
 
 // Asummes movements in descending order of creation
 function useConvertedMovements(movements) {
   const { currentTermId } = useCourtClock()
 
-  const effectiveStates = movements.map(mov =>
-    isMovementEffective(mov, currentTermId)
-  )
+  const effectiveStates = movements
```

Comment:
Can `movements` be falsy? It comes from here if I’m not mistaking: https://github.com/aragon/court-dashboard/pull/71/files#diff-30a133659d1207cf561a6e864720abecR87-R91

Reply:
It can be, see here https://github.com/aragon/court-dashboard/blob/ddfac8c289c4c27ce84b3936beb90fb2805cef16/src/hooks/subscription-hooks.js#L61

This is the case when we are still fetching balances

We only set the movements as empty array if we are done fetching and the user doesn't have any 24hs movements

Reply:
Oh right I missed that, thanks for the explanation!